### ISSUE TITLE
auto connection management does not work in 5.0.0-beta.14(mysql)

### ISSUE THREAD

**Issue by SangJinny** [sequelize/sequelize#10177]
I am using 5.0.0-beta.14, and I found serious problem in connection pool. I think managing connection pool is not working properly.

here is my code.

```js
'use strict';

var fs        = require('fs'),
	path      = require('path'),
	Sequelize = require('sequelize'),
	basename  = path.basename(__filename),
	env       = process.env.NODE_ENV || 'develop',
	config    = require('../../config/database.json')[env],
	db        = {};

config.password = process.env.RDB_PW;

var sequelize = new Sequelize(config.database, config.username, config.password, config);

fs.readdirSync(__dirname)
	.filter(file => {
		return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
	})
	.forEach(file => {
		var model = sequelize['import'](path.join(__dirname, file));
		db[model.name] = model;
	});

Object.keys(db).forEach(modelName => {
	if (db[modelName].associate) {
		db[modelName].associate(db);
	}
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
```

When I use 5.0.0-beta.13, I could see the log that auto-connection management is working. I think it should be working even db access is not occurred. see this.

![kakaotalk_photo_2018-11-19-09-01-19](https://user-images.githubusercontent.com/12556691/48679980-b7454180-ebd9-11e8-866a-a0970542ae7c.png)
(at that time. no db access is occurred by application.)

But in 5.0.0-beta.14, it does not work. 
when db access occurred, connection managing is work. but does not work in idle.

__Dialect: mysql2
__Dialect version: 1.5.3
__Database version: mysql 5.7.17
__Sequelize version: 5.0.0-beta.14
__Tested with latest release: YES (5.0.0-beta.14)

**Comment by SangJinny** [comment]
```
"develop" : {
    "database": "develop",
    "username": "deploy",
    "host": "host,
    "dialect": "mysql",
    "timezone": "+09:00",
    "pool": {
      "min": 5,
      "max": 20,
      "idle": 10000, // refresh every 10 idle second!!
      "evict": 10000
    },
    "define": {
      "underscored": true
    }
  }
```

You can see my connection pool settings.
I think sequelize should refresh connection every 10 idle second. and I found it works in 5.0.0-beta13 (I showed you log dump)
and in 5.0.0-beta14, no refresh occur.

thank you!

**Comment by sushantdhiman** [comment]
Sequelize with beta.14 will exit process if there is nothing more to do i.e. event loop is empty. In past versions like beta.13 pool wont close even if there was nothing more to do.

I think what you mean idle is process exit, sequelize will exit itself after 10th sec. Also after 10th second pool will remove that old connection you will get new connection if new query is made, that I've verified several times.

**Comment by SangJinny** [comment]
![image](https://user-images.githubusercontent.com/12556691/48690243-170b0f00-ec11-11e8-9915-710b5c53b361.png)

I think beta.14's removing may be not work properly.
that graph is my application's db status(connection) graph.

I used beta14 in 11/07. and the number of sleep-state-connection is increased.
I don't know why it did. but when I downgrade to beta13(in 11/15), the number of connection is maintained.

P.S. I found your saying is right. beta14 removes old connection when application get new db request. but I don't know why that graph features.

thank you!

**Comment by sushantdhiman** [comment]
Thanks for this report, I'll investigate. May be we are not closing connections properly?

**Comment by SangJinny** [comment]
right. in my sequelize setting code. there is no connection closing logic. because I know that sequelize manange connection automatically itself.

so sequelize(beta.14) may have a problem in closing connection.

thank you for your favorable comment.

**Comment by sushantdhiman** [comment]
@salsa2k Certainly not related as there's been no change whatsoever related to pooling in recent v4 versions

**Comment by sushantdhiman** [comment]
Hi @SangJinny I tried to reproduce this, but I discovered pooling is working properly, under no condition sequelize used more connections than what defined by `max`.

Do you think your server is configured to use permanent connections as mysql2 by default does not create permanent connections and they should automatically idle out?

**Comment by SangJinny** [comment]
Hi, Thank you for your investigation.
As I analyse my DB monitering graph, I found that **DB Read and Insert access were increased** in that time(using beta 14). It may help you.

And I also found I used **generic-pool ^3.4.2** (sequelize5 uses sequelize-pool and it forked generic-pool 2.5)
I think it may be reason of this problem. I should do test about it.(run application not using generic-pool ^3.4.2) and I will notice you about my test.
